src/controllers/users.controller.ts

```diff
@@ -0,0 +1,71 @@
+import {ValidatedRequest} from 'express-joi-validation';
+import * as userValidation from '../validation/validation-schemas';
+import UserService from '../services/user.service';
+import {Request, Response} from 'express';
+import {RESPONSE_MESSAGES} from '../config/messages';
+
+export class UsersController {
+  static async getSuggestedUsers(
```

Comment:
every exception crashes the application. This is because your handler is async. Even though express supports async function, it cannot catch error thrown from async function
you need to add try catch in every controller method. And, if error is occurred, use next(err) to pass it to the error handle middleware.

For example
```ts
  static async getSuggestedUsers(
    req: ValidatedRequest<userValidation.UserRequestBodySchema>,
    res: Response,
    next: NextFunction
  ) {
    try {
      const loginSubstring = req.query.loginSubstring || '';
      const limit = req.query.limit ? +req.query.limit : undefined;
      const suggestedUsers = await UserService.getSuggestedUsers(
        loginSubstring,
        limit
      );
      res.send(suggestedUsers);
    } catch (error) {
      next(error);
    }
  }
```

You need to do the same for every method

Reply:
added try-catch everywhere in controller